src/AppInstallerCLICore/Workflows/PortableInstallFlow.cpp

```diff
@@ -0,0 +1,379 @@
+            uninstallEntry.SetValue(PortableValueName::InstallDate, Utility::GetCurrentDateForARP());
+            uninstallEntry.SetValue(PortableValueName::URLInfoAbout, manifest.DefaultLocalization.Get<Manifest::Localization::PackageUrl>());
+            uninstallEntry.SetValue(PortableValueName::HelpLink, manifest.DefaultLocalization.Get<Manifest::Localization::PublisherSupportUrl>());
+            uninstallEntry.SetValue(PortableValueName::UninstallString, L"winget uninstall --product-code " + productCode);
+            uninstallEntry.SetValue(PortableValueName::WinGetInstallerType, ConvertToUTF16(InstallerTypeToString(InstallerTypeEnum::Portable)));
+            uninstallEntry.SetValue(PortableValueName::WinGetPackageIdentifier, manifest.Id);
+            uninstallEntry.SetValue(PortableValueName::WinGetSourceIdentifier, sourceIdentifier);
+            uninstallEntry.SetValue(PortableValueName::PortableTargetFullPath, GetPortableTargetFullPath(context).wstring());
+            uninstallEntry.SetValue(PortableValueName::PortableSymlinkFullPath, GetPortableSymlinkFullPath(context).wstring());
+            uninstallEntry.SetValue(PortableValueName::SHA256, Utility::SHA256::ConvertToWideString(context.Get<Execution::Data::HashPair>().second));
+            uninstallEntry.SetValue(PortableValueName::InstallLocation, GetPortableTargetDirectory(context).wstring());
+            AICLI_LOG(CLI, Info, << "Writing to Uninstall registry complete.");
+        }
+
+        void MovePortableExeAndCreateSymlink(Execution::Context& context)
+        {
+            const std::filesystem::path& installerPath = context.Get<Execution::Data::InstallerPath>();
+            const std::filesystem::path& targetFullPath = GetPortableTargetFullPath(context);
+            const std::filesystem::path& symlinkFullPath = GetPortableSymlinkFullPath(context);
+            const std::filesystem::path& targetDirectory = GetPortableTargetDirectory(context);
+
+            bool isDirectoryCreated = false;
+            if (!std::filesystem::exists(targetDirectory) || !std::filesystem::is_directory(targetDirectory))
+            {
+                if (!std::filesystem::create_directories(targetDirectory))
+                {
+                    AICLI_LOG(CLI, Info, << "Unable to create the target install directory: " << targetDirectory);
+                    AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_PORTABLE_FAILED_TO_CREATE_DIRECTORY);
+                }
+
+                AICLI_LOG(CLI, Info, << "Created target install directory: " << targetDirectory);
+                isDirectoryCreated = true;
+            }
+
+            Portable::PortableARPEntry uninstallEntry = Portable::PortableARPEntry(
+                ConvertToScopeEnum(context.Args.GetArg(Execution::Args::Type::InstallScope)),
+                context.Get<Execution::Data::Installer>()->Arch,
+                ConvertToUTF16(GetPortableProductCode(context)));
```

Comment:
This is why `GetPortableProductCode` should `MakeSuitablePathPart`, even though it is more restrictive than the registry requires.
```
Key names cannot include the backslash character (\), but any other printable character can be used.
```

Reply:
Added MakeSuitablePathPart to GetPortableProductCode